kolibri/plugins/coach/assets/src/state/actions/reports.js

```diff
@@ -26,22 +26,99 @@ const ChannelResource = coreApp.resources.ChannelResource;
 const FacilityUserResource = coreApp.resources.FacilityUserResource;
 
 
+// helper function for showRecent, provides list of channels with recent activity
 function _showRecentChannels(store, classId) {
-  store.dispatch('CORE_SET_PAGE_LOADING', true);
-  store.dispatch('SET_PAGE_NAME', Constants.PageNames.RECENT);
-  const channelPromise = ChannelResource.getCollection();
-  channelPromise.fetch().then(
-    channels => {
+  function __getChannelLastActive(channel) {
+    // helper function for _showRecentChannels
+    // @param channel to get recentActivity for
+    // @returns promise that resolves channel with lastActive value in object:
+    // {
+    //   'channelId': {
+    //     amount: 'int',
+    //     measure: 'month or day',
+    //   },
+    // }
+    function ___timePassedSince(lastActiveTime) {
+      // helper function for __getChannelLastActive
+      // @param lastActiveTime --  date in string format
+      // @returns object representing time passed since input in days or months:
+      // {
+      //   amount: 'int',
+      //   measure: 'month or day',
+      // }
+      const dayMeasure = (ms) => Math.round(ms / (8.64e+7));
+      const monthMeasure = (ms) => Math.round(ms / (2.628e+9));
+
+      const currentDate = new Date();
+      const lastActiveDate = new Date(lastActiveTime);
+      // subtracting dates returns time interval in milliseconds
+      const millisecondsEllapsed = currentDate - lastActiveDate;
+
+      const monthsAgo = monthMeasure(millisecondsEllapsed);
+      // returns months amount of days has surpassed a month
+      if (monthsAgo) {
+        return {
+          amount: monthsAgo,
+          measure: 'month',
+          raw: millisecondsEllapsed,
+        };
+      }
+      // and days otherwise
+      return {
+        amount: dayMeasure(millisecondsEllapsed),
+        measure: 'day',
+        raw: millisecondsEllapsed,
+      };
+    }
+    const summaryPayload = {
+      channel_id: channel.id,
+      collection_kind: ReportConstants.UserScopes.FACILITY,
+      collection_id: store.state.core.session.facility_id,
+    };
+
+    // workaround for conditionalPromise.then() misbehaving
+    return new Promise(
```

Comment:
Workaround for `ConditionalPromise.then()` not returning a promise with the new resolved value. It gets a little messy, but I wanted to do it this way because `channel_id` is alreadly included inside of this helper function, and `ContentSummaryResource` doesn't resolve the `channel_id` -- it resolves `content_id` and a `pk` somewhere, but none of the `id`s included were the channel's.